mitmproxy/protocol/http.py

```diff
@@ -368,7 +375,10 @@ def establish_server_connection(self, flow):
                 self.set_server(address, tls, address.host)
             # Establish connection is neccessary.
             if not self.server_conn:
-                self.connect()
+                try:
+                    self.connect()
+                except:
```

Comment:
If the connect fails - we need some way of passing this up the call stack.
Returning True and False seems like a clean solution.